core/modules/util/threadSafe.h

```diff
@@ -73,21 +73,23 @@ class Flag {
 
     /** Sets flag value to 'val' and returns the old value of flag.
      */
-    virtual T set(T val) {
+    virtual T exchange(T val) {
```

Comment:
Flag is serving a specific purpose. There is a case where std::atomic is what is needed most of the time, but you cannot access the underlying locking mechanism, and occasionally you have to have access to the mutex or there's no way prevent a race condition. Flag looks like an atomic, but you can get at the mutex if you need to. The example of this is Executive::_cancelled.